### ISSUE TITLE
Make users confirm clicking out of a flow, replace X button with Cancel buttons

### ISSUE THREAD

**Issue by evvvritt** [drips-network/app#656]

**Comment by evvvritt** [comment]
hmm, @efstajas does your modal system allow another (small) modal to appear above the current modal? if not, perhaps that's rolled into the new alert + small modal issue (#646)

**Comment by brandonhaslegs** [comment]
BTW not every modal needs a "confirm close" alert when you click outside of it. Only if you're in the middle of a flow.

**Comment by efstajas** [comment]
> hmm, @efstajas does your modal system allow another (small) modal to appear above the current modal? if not, perhaps that's rolled into the new alert + small modal issue (#646)

nope, it doesn't atm. Would need some refactoring to do that effectively....

**Comment by efstajas** [comment]
@brandonhaslegs Could you consolidate here the design for the step that makes you confirm closing out, the cancel buttons, and define the behavior in detail (when should it ask you to confirm etc.)

**Comment by brandonhaslegs** [comment]
I'm not going to be able to catch every modal (every modal needs to change) but I can try.

- [ ] Collect funds - just remove the X button, there's already a "Never mind" button to close it
- [ ] Add funds - already has a cancel button, remove X
- [ ] Add USDC (token) - add cancel button , remove X
- [ ] 
- [ ] Create stream
   - [ ] there's already a cancel button, so the X can be removed.
   - [ ] once the user has entered text into any field on the first step, show the dialog confirming they want to stop the flow
   - [ ] "You sure?" and "You will lose any progress you made" could work for generic text
- [ ] Edit stream - there's already a cancel button, so the X can be removed
- [ ] Pause stream - this needs a cancel button. it only has a back button
- [ ] Claim your project - add cancel button (and apply the close logic from above, ie if the user has entered anything, then show the close dialog, if not, let them close with no dialog)
- [ ] Edit project details (emoji/image and color) - just add cancel button, remove X
- [ ] Edit splits - already has a cancel button, remove X
- [ ] Donate instantly - already has a cancel button, remove X
- [ ] Add to a Drip List - already has a cancel button, remove X
- [ ] _(Note: somehow I came across this locked in Drip List creation flow but I can't find it anymore)_ Create a Drip List - this still needs to be made into a modal (until we redesign it and change it to it's own page-based flow. #1303 But the state it's in now is just completely hurt. literally a trap you can't get out of

There's probably some I missed

Also note: when this is built, add clicking outside a modal to close it as standard functionality for all modals. Same logic applies to this, if you've entered anything into any field, it should warn you before closing. If not, just close.

**Comment by efstajas** [comment]
@brandonhaslegs pls add design for warning modal, clarify behavior (esc, click out of modal etc.)

**Comment by brandonhaslegs** [comment]
![Image](https://github.com/user-attachments/assets/a94e1afd-74fa-470a-bb2f-4f2ba6570778)
